luigi_pipeline/lib/hail_tasks.py

```diff
@@ -25,6 +27,33 @@ class HailMatrixTableTask(luigi.Task):
     """
     source_paths = luigi.ListParameter(description='List of paths to VCFs to be loaded.')
     dest_path = luigi.Parameter(description='Path to write the matrix table.')
+    genome_version = luigi.Parameter(description='Reference Genome Version (37 or 38)')
+
+    @staticmethod
+    def hl_mt_impute_sample_type(mt, genome_version, threshold=0.3):
+        """
+        Impute the sample type by chcking against a list of common coding and non-coding variants.
```

Comment:
Sorry took a while to get to this.. the logic LGTM.  Re. method boundaries and names,
- it's confusing that the "impute-sample-type" method returns counts rather than the sample type, and the actual decision about sample type happens elsewhere. I'd vote for putting the code from line 39-61 (https://github.com/macarthur-lab/hail-elasticsearch-pipelines/pull/128/files#diff-7087d398dbeca709f67c00788f5802deR39) inside this method, and have the return value be "WES" or "WSG". 

- also would call this `impute_sample_type` instead of `hl_mt_impute_sample_type` since it's already in a HailMatrixTableTask class.

Reply:
Good point on the naming. I thought it was weird as well, and considered having the function return a type ('WGS', 'WES', 'unknown'). However, I don't think the validation logic (throwing errors) should be included there, since this is seqr specific. I'll just rename it `sample_type_stats`.
And I remembered having a reason for the `hl_mt` prefix, but can't remember it now. I'll change it and see if I remember later.